[BUG] loading LZ4 Parquet files SIGSEGs

**Describe the bug**
Buried in the CI runs like #12439 run is another SIGSEGV however the JVM is not given a chance to generate hs_err* file. Here the backtrace from cuda-gdb on the core file.

```
#1  0x0000790149ea5139 in malloc () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x0000790148aad074 in __cxa_allocate_exception () from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x000078fc821ed86b in cudf::io::parquet::detail::(anonymous namespace)::from_parquet_compression(cudf::io::parquet::detail::Compression) [clone .cold] () from /tmp/cudf4415370315557527554.so
...
#9103 0x000078fc821eddb5 in cudf::io::parquet::detail::(anonymous namespace)::decompress_page_data(cudf::host_span<cudf::io::parquet::detail::ColumnChunkDesc const, 18446744073709551615ul>, cudf::host_span<cudf::io::parquet::detail::PageInfo, 18446744073709551615ul>, cudf::host_span<cudf::io::parquet::detail::PageInfo, 18446744073709551615ul>, rmm::cuda_stream_view) [clone .cold] () from /tmp/cudf4415370315557527554.so
#9104 0x000078fc83310aa3 in cudf::io::parquet::detail::reader::impl::setup_next_subpass(cudf::io::parquet::detail::reader::impl::read_mode) () from /tmp/cudf4415370315557527554.so
#9105 0x000078fc832f0808 in cudf::io::parquet::detail::reader::impl::read_chunk() () from /tmp/cudf4415370315557527554.so
#9106 0x000078fc832e81ae in cudf::io::parquet::detail::chunked_reader::read_chunk() const () from /tmp/cudf4415370315557527554.so
#9107 0x000078fc831320f0 in cudf::io::chunked_parquet_reader::read_chunk() const () from /tmp/cudf4415370315557527554.so
#9108 0x000078fc82686f88 in Java_ai_rapids_cudf_ParquetChunkedReader_readChunk () from /tmp/cudf4415370315557527554.so
```

**Steps/Code to reproduce bug**

Simplified  `repro.py`

```Python
from pyspark.sql import SparkSession
spark = (
    SparkSession.builder.getOrCreate()
)
spark.read.parquet('thirdparty/parquet-testing/data/non_hadoop_lz4_compressed.parquet').show() 
```

```
TZ=UTC ~/dist/spark-3.5.5-bin-hadoop3/bin/spark-submit \
  --packages com.nvidia:rapids-4-spark_2.12:25.06.0-SNAPSHOT \
  --conf spark.jars.repositories=$YOUR_MAVEN_URL \
  --conf spark.plugins=com.nvidia.spark.SQLPlugin \
  --conf spark.rapids.sql.explain=ALL \
  --driver-java-options -Dai.rapids.cudf.preserve-dependencies=true \
  repro.py
...
25/04/07 00:58:26 INFO GpuParquetMultiFilePartitionReaderFactory: Using the coalesce multi-file Parquet reader, files: file:///home/gshegalov/gits/NVIDIA/spark-rapids/thirdparty/parquet-testing/data/non_hadoop_lz4_compressed.parquet task attemptid: 1
25/04/07 00:58:26 WARN MultiFileReaderThreadPool: Configuring the file reader thread pool with a max of 64 threads instead of spark.rapids.sql.multiThreadedRead.numThreads = 20
Segmentation fault (core dumped)
```

#### Upstream CPU Spark 
Vanilla Apache Spark without the Plugin fails to read this file as well but it does not crash the process 
```
Caused by: java.lang.ArrayIndexOutOfBoundsException
        at org.apache.hadoop.io.compress.lz4.Lz4Decompressor.setInput(Lz4Decompressor.java:101)
        at org.apache.hadoop.io.compress.BlockDecompressorStream.decompress(BlockDecompressorStream.java:104)
        at org.apache.hadoop.io.compress.DecompressorStream.read(DecompressorStream.java:105)
        at java.io.DataInputStream.readFully(DataInputStream.java:195)
        at java.io.DataInputStream.readFully(DataInputStream.java:169)
        at org.apache.parquet.bytes.BytesInput$StreamBytesInput.toByteArray(BytesInput.java:286)
        at org.apache.parquet.bytes.BytesInput.toByteBuffer(BytesInput.java:237)
        at org.apache.parquet.bytes.BytesInput.toInputStream(BytesInput.java:246)
        at org.apache.parquet.column.values.dictionary.PlainValuesDictionary$PlainLongDictionary.<init>(PlainValuesDictionary.java:154)
        at org.apache.parquet.column.Encoding$1.initDictionary(Encoding.java:96)
        at org.apache.parquet.column.Encoding$5.initDictionary(Encoding.java:163)
        at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.<init>(VectorizedColumnReader.java:123)
        at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.initColumnReader(VectorizedParquetRecordReader.java:437)
        at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.checkEndOfRowGroup(VectorizedParquetRecordReader.java:427)
        at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:335)
        at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
        at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:131)
        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:286)
        ... 23 more
```` 

#### 25.02.1 Behavior

The test works in the prior release `25.02.1` because the Plugin would fail to load this file gracefully 
```
Caused by: ai.rapids.cudf.CudfException: CUDF failure at: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni-release-18-cuda11/thirdparty/cudf/cpp/src/io/parquet/reader_impl_chunking.cu:787: Unsupported compression type
        at ai.rapids.cudf.ParquetChunkedReader.readChunk(Native Method)
        at ai.rapids.cudf.ParquetChunkedReader.readChunk(ParquetChunkedReader.java:200)
        at com.nvidia.spark.rapids.ParquetTableReader.$anonfun$next$1(GpuParquetScan.scala:2918)
        ... 50 more
```

**Expected behavior**
should throw an exception like CPU without crashing

**Environment details (please complete the following information)**
any 

**Additional context**
LZ4 support

--- a/integration_tests/src/main/python/parquet_testing_test.py
+++ b/integration_tests/src/main/python/parquet_testing_test.py
@@ -62,13 +62,6 @@ _xfail_files = {
     "hadoop_lz4_compressed_larger.parquet": "cudf does not support Hadoop LZ4 format",
     "nested_structs.rust.parquet": "PySpark cannot handle year 52951",
 }
-
-_skip_hard_crash_files = {
-    "hadoop_lz4_compressed.parquet": "Avoiding SIGSEGV https://github.com/NVIDIA/spark-rapids/issues/12453",
-    "hadoop_lz4_compressed_larger.parquet": "Avoiding SIGSEGV https://github.com/NVIDIA/spark-rapids/issues/12453",
-    "non_hadoop_lz4_compressed.parquet": "Avoiding SIGSEGV https://github.com/NVIDIA/spark-rapids/issues/12453",
-}
-
 if is_before_spark_330():
     _xfail_files["rle_boolean_encoding.parquet"] = "Spark CPU cannot decode V2 style RLE before 3.3.x"
 
@@ -144,14 +137,9 @@ def locate_parquet_testing_files():
 def gen_testing_params_for_errors():
     result = []
     for f in locate_parquet_testing_files():
-        basename = os.path.basename(f)
-        error_obj = _error_files.get(basename, None)
+        error_obj = _error_files.get(os.path.basename(f), None)
         if error_obj is not None:
-            skip_reason = _skip_hard_crash_files.get(basename, None)
-            if skip_reason is None:
-                result.append((f, error_obj))
-            else:
-                result.append(pytest.param(f, error_obj, marks=pytest.mark.skip(reason=skip_reason)))
+            result.append((f, error_obj))
     return result
 
 def gen_testing_params_for_valid_files():
@@ -160,16 +148,11 @@ def gen_testing_params_for_valid_files():
         basename = os.path.basename(f)
         if basename in _error_files:
             continue
-        skip_reason = _skip_hard_crash_files.get(basename, None)
-        if skip_reason:
-            files.append(pytest.param(f, marks=pytest.mark.skip(reason=skip_reason)))
+        xfail_reason = _xfail_files.get(basename, None)
+        if xfail_reason:
+            files.append(pytest.param(f, marks=pytest.mark.xfail(reason=xfail_reason)))
         else:
-            xfail_reason = _xfail_files.get(basename, None)
-            if xfail_reason:
-                files.append(pytest.param(f, marks=pytest.mark.xfail(reason=xfail_reason)))
-            else:
-                files.append(f)
-
+            files.append(f)
     return files
 
 @pytest.mark.parametrize("path", gen_testing_params_for_valid_files())

